### PR TITLE
[GPU] dump verbose log to file

### PR DESCRIPTION
**Pull request by isanghao**
### Details:
- New debug config for dumping verbose log into file

### CODE REVIEW

File: src/plugins/intel_gpu/src/runtime/debug_configuration.cpp [hunk 1]
```diff
@@ -128,6 +130,7 @@ static void print_help_messages() {
     message_list.emplace_back("OV_GPU_Help", "Print help messages");
     message_list.emplace_back("OV_GPU_Verbose", "Verbose execution");
     message_list.emplace_back("OV_GPU_VerboseColor", "Print verbose color");
+    message_list.emplace_back("OV_GPU_VerboseFile", "Filename to dump verbose log");
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/runtime/debug_configuration.cpp:133]: Why do we need that? You can just redirect cerr and cout to the file if you want, i.e.
```
$ OV_VERBOSE=4 ./my_app &> log.txt
```

Comment (isanghao) [reply src/plugins/intel_gpu/src/runtime/debug_configuration.cpp:133]: When application is launched from cmdline, it is possible. But in some cases, it is launched from windows application directly. I think that is why clintercept has similar option.

File: src/plugins/intel_gpu/src/runtime/debug_configuration.cpp [hunk 2]
```diff
@@ -318,6 +323,14 @@ debug_configuration::debug_configuration()
         exit(0);
     }
 
+    if (verbose_file.length() > 0) {
+            static std::ofstream fout;
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/runtime/debug_configuration.cpp:327]: nit: something is wrong with indentation

**Review by vladimir-paramuzov — APPROVED** [review]

File: src/plugins/intel_gpu/src/runtime/debug_configuration.cpp (hunk 2 above)
Comment (isanghao) [reply src/plugins/intel_gpu/src/runtime/debug_configuration.cpp]: fixed, thanks

**Pull request merged**